huaweicloud/services/dli/resource_huaweicloud_dli_database.go

```diff
@@ -88,12 +103,28 @@ func resourceDliSQLDatabaseCreate(ctx context.Context, d *schema.ResourceData, m
 		EnterpriseProjectId: common.GetEnterpriseProjectID(d, cfg),
 		Tags:                utils.ExpandResourceTags(d.Get("tags").(map[string]interface{})),
 	}
-	respBody, err := databases.Create(c, opts)
+
+	resp, err := databases.Create(client, opts)
 	if err != nil {
-		return diag.Errorf("error creating DLI database, %s", err)
-	}
-	if !respBody.IsSuccess {
-		return diag.Errorf("unable to create the database: %s", respBody.Message)
+		if _, ok := err.(golangsdk.ErrDefault408); !ok {
+			return diag.Errorf("error creating DLI database (%s): %s", dbName, err)
+		}
+
+		// Return synchronization job result times out.
+		// At this time, the job has entered the creation phase.
+		stateConf := &resource.StateChangeConf{
+			Pending:      []string{"PENDING"},
+			Target:       []string{"COMPLETED"},
+			Refresh:      databaseCreateRefreshFunc(client, dbName),
+			Timeout:      d.Timeout(schema.TimeoutCreate),
+			PollInterval: 10 * time.Second,
+		}
+		_, err = stateConf.WaitForStateContext(ctx)
+		if err != nil {
+			return diag.Errorf("error creating DLI database (%s): %s", dbName, err)
+		}
+	} else if resp != nil && !resp.IsSuccess {
+		return diag.Errorf("the request was sent successfully, but some errors occurred: %s", resp.Message)
```

Comment:
if resp == nil {
  return diag.Errorf("error creating DLI database (%s): The creation API response is empty.", dbName)
}

if !resp.IsSuccess {
  return diag.Errorf("the request was sent successfully, but some errors occurred: %s", resp.Message)
}

Reply:
we don't care about the case of the response is nil, because we don't need to parse it.